packages/entity-server/src/routes/hierarchy/middleware/fields.ts

```diff
@@ -46,20 +49,41 @@ export const extractFieldsFromQuery = (queryFields?: string) => {
 };
 
 type Writable<T> = { -readonly [field in keyof T]?: T[field] };
+
+const assignExtendedFieldToResponseObject = async (
```

Comment:
It's amazing how functionally oriented you've become considering your Java background! This is totes cool, but my instinct would be to create some sort of a `EntitiesResponse` class that held on to `mappedEntities` and had `assignField` and `assignExtendedField` methods. At the end you could call `getBuiltResponse`. I just don't like passing variables in to functions with the idea that they'll be mutated by that function - feels off.

Reply:
You're totally right, that's a much cleaner pattern!

Reply:
I'm so glad that you requested this! Since refactoring to use a class I've been able to finally crack the generics problem I was facing in this code area that was forcing me to do a really ugly explicit typecast previously. Tbh I'm still not entirely sure why it works now but not previously, but I am very happy it does 🎉